src/lib/utils/fixFingerprintDataTypes.ts

```diff
@@ -2,23 +2,21 @@ import { IFingerprint } from "../../types";
 import { fingerprintNoQuotes } from "../../types/Fingerprint";
 
 export function fixFingerprintDataTypes(fingerprint: IFingerprint): IFingerprint {
```

Comment:
Let's move those fixes to zod's preprocesses similar to how you made the `zodDateString` method.